app-thunderbird/src/test/kotlin/net/thunderbird/android/unread/UnreadWidgetDataProviderTest.kt

```diff
@@ -0,0 +1,150 @@
+package net.thunderbird.android.unread
+
+import android.content.Context
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import com.fsck.k9.Account
+import com.fsck.k9.Preferences
+import com.fsck.k9.controller.MessageCounts
+import com.fsck.k9.controller.MessageCountsProvider
+import com.fsck.k9.mailstore.Folder
+import com.fsck.k9.mailstore.FolderRepository
+import com.fsck.k9.mailstore.FolderType
+import com.fsck.k9.search.SearchAccount
+import com.fsck.k9.ui.folders.FolderNameFormatter
+import com.fsck.k9.ui.messagelist.DefaultFolderProvider
+import com.fsck.k9.widget.unread.UnreadWidgetConfiguration
+import com.fsck.k9.widget.unread.UnreadWidgetDataProvider
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.test.AutoCloseKoinTest
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class UnreadWidgetDataProviderTest : AutoCloseKoinTest() {
```

Comment:
I've created #7781 so we don't need to have multiple copies of this test.

Reply:
I'll do this in a follow up pull-request.